### ISSUE TITLE
USB C display causes generic display to crash [BUG]

### ISSUE THREAD

**Issue by carelw** [zacmks/Hackintosh-Aero-15X#29]
**Describe the bug**
When booting with a USB type-C display connected (mini DP doesn't seem to work and HDMI doesn't work due to external graphics card connectivity) the main laptop display seems to crash.

**To Reproduce**
1. Boot with USB type-C display connected
2. Screen seems to be zoomed in whilst booting
3. Apple boot screen starts loading on main laptop screen, then stops working (goes black) and screen appears on the connected USB type-C display

**Desktop (please complete the following information):**
 - OS: Big Sur
 - Version latest supported

**Comment by brightplastik** [comment]
Hello. Same issue here too. It's a well known problem with the CPU family, and to my knowledge nobody has solved the issue. 
BUT! 
Easily enough, wait for the pc to fully bootup. Modify the OS to NOT go to sleep when you close the lid (google it).
When you apply this setting, the OS will move all your active windows on the external display by closing the lid. 
Wait a few seconds, and by reopening it the pc will finally output on both screens, just a few seconds after the bootup (*).

*someone was suggesting the display eventually lights up after some minutes. It does not with my pc, but in exchange this works without waiting.

**Comment by zacmks** [comment]
There's another workaround as well:

1. Go to:
System Preferences -> Displays -> Arrangement.

2. There, tick twice the checkbox "Mirror Displays" (enable/disable)
![image](https://user-images.githubusercontent.com/14233911/144792243-2738a831-ee04-4df1-9c7c-4252adbb403d.png)
 
3. Then, your internal display should work fine.

*I do suggest keeping your external (USB-C) monitor as the main monitor (so you can do this operation after reboots)

If anyone could find the root cause of the issue (or even a fix for it), I'll be glad.
I tried several video + framebuffers + connectors configurations but it didn't work.

**Comment by brightplastik** [comment]
Unfortunately this workaround does not work for me, at least with my current devices/settings/kexts. The builtin screen is turned on, as I can see the backlight, but signal does not come after clicking two times on "mirror displays". Close-opening the lid does work. 
My external display is an old VGA, plugged in through a Dell TB16 TB3 dock. 
VGA is an old and rudimental protocol, that's why, maybe.
I cannot test right now, but maybe a different external signal (being either HDMI or DVI or else) could surely wake up the builtin screen, with your method.

My two cents, purely for debugging purpose.

**Comment by zacmks** [comment]
> Unfortunately this workaround does not work for me, at least with my current devices/settings/kexts. The builtin screen is turned on, as I can see the backlight, but signal does not come after clicking two times on "mirror displays". Close-opening the lid does work. My external display is an old VGA, plugged in through a Dell TB16 TB3 dock. VGA is an old and rudimental protocol, that's why, maybe. I cannot test right now, but maybe a different external signal (being either HDMI or DVI or else) could surely wake up the builtin screen, with your method.
> 
> My two cents, purely for debugging purpose.

Yes that could be the case. VGA uses analog signals which in my point of view should be harder to send/receive 'commands', considering the nature of analog signals (compared to the DVI/HDMI digital interfaces). I have a simple USB-C to HDMI adapter. If I get my hands on an VGA monitor + adapter I'll test here as well.

Imo this is the worst 'maybe solvable' issue of this hackintosh setup (not considering unsupported dGPU here), personally the machine that I use performs better compared to actual Mac machines hahaha
